model/roles.go

```diff
@@ -64,11 +64,12 @@ type RoleManifest struct {
 
 // JobReference represents a job in the context of a role
 type JobReference struct {
-	*Job              `yaml:"-"`                 // The resolved job
-	Name              string                     `yaml:"name"`    // The name of the job
-	ReleaseName       string                     `yaml:"release"` // The release the job comes from
-	ExportedProviders map[string]jobProvidesInfo `yaml:"provides"`
-	ResolvedConsumers map[string]jobConsumesInfo `yaml:"consumes"`
+	*Job                `yaml:"-"`                 // The resolved job
+	Name                string                     `yaml:"name"`    // The name of the job
+	ReleaseName         string                     `yaml:"release"` // The release the job comes from
+	ExportedProviders   map[string]jobProvidesInfo `yaml:"provides"`
+	ResolvedConsumers   map[string]jobConsumesInfo `yaml:"consumes"`
+	ContainerProperties *JobContainerProperties    `yaml:"properties"`
```

Comment:
Yes, this was cruft from earlier attempts. I removed the pointers from this and its children. I was briefly confused why some tests broke, but it was because `ValidateExposedPorts` doesn't just validate, it handles some legacy stuff.

Reply:
Fixed in d8acbdc